### DISCUSSION TITLE
Exception [The semaphore timeout period has expired]

### DISCUSSION THREAD

**Discussion by KotobaTrilius** [Rapptz/discord.py#9221] [Q&A]
I am trying to get start with a bot, and I try the example code as below:
```
intents = discord.Intents.default()
intents.message_content = True

client = discord.Client(intents=intents)

@client.event
async def on_ready():
    print(f'We have logged in as {client.user}')

@client.event
async def on_message(message):
    if message.author == client.user:
        return

    if message.content.startswith('$hello'):
        await message.channel.send('Hello!')

client.run(TOKEN)
```

After I run the code, the command line shows:
```
2023-02-05 02:29:44 INFO     discord.client logging in using static token
```

and seconds later ends in an exception:
```
aiohttp.client_exceptions.ClientConnectorError: Cannot connect to host discord.com:443 ssl:default [The semaphore timeout period has expired]
```

Working on Windows 10, with Python 3.9,
I have googled but I got no solution, and I have tried resetting the token for many times which still doesn't work.
**Many have similar exceptions but mine is "[The semaphore timeout period has expired]" which I can find no useful solution.**
How can I fix it?

**Comment by WilliamsHoward** [comment]
@KotobaSuke Hey good day , I'm meeting the same issue here with same error message , the token is valid , not sure if you've already solved the problem , if so , could you share some insights on how you fixed this error? appreciate your help !

**Comment by wekul** [comment]
This is not an issue with your token. This is an issue with your connection to discord, wether that is due to your internet or wherever you are hosting the bot.

**Comment by WilliamsHoward** [comment]
I think @lukewain got this right , yesterday I finally found out it is due to the proxy of my local , if it turns out your issue is also caused by the proxy @KotobaSuke , you can try to set the proxy with your local proxy url when creating the bot, like below for your reference:
`bot = commands.Bot(command_prefix="xx", intents=x, heartbeat_timeout=x, proxy='http://xxx.xx.xx.x:xxxx')`
